### ISSUE TITLE
ESP key generation, generate filename based on device mac address (if not specified)

### ISSUE THREAD

**Issue by tripflex** [cesanta/mos-tool#36]
After talking about this on Gitter wanted to open an issue regarding this.

During key generation for ESP32 devices, you have to specify a filename for the key file.  May be a good feature to add to auto generate the key filename (only the filename `XXXX.key`) based on the MAC address, if user does not supply one (or possibly using replace, like in mos with `?`):

Something like this:
```shell
mos -X esp32-gen-key flash_encryption_key fe_??????.key --esp32-enable-flash-encryption --dry-run=false
```

Which would result in `fe_A123B123.key` or something similar

https://github.com/cesanta/mos-tool/blob/master/mos/esp32_gen_key.go

**Comment by rojer** [comment]
makes sense. PRs welcome!

**Comment by gadget-man** [comment]
I would love this. Currently I'm doing a really long winded and inefficient process to create a serial number based on the Mac address using a bash script. It would be awesome if the key file could be specified to be based one the Mac address like we have elsewhere.

**Comment by rojer** [comment]
this has been released in mos-latest 201903041227

Pro tip: `export MOS_ESP32_ENCRYPTION_KEY_FILE=/path/to/keys/fe-??????.key.bin` will make it completely seamless - when enabling encryption, new keys will be stored in `/path/to/keys` automatically and when flashing, keys will be found there if needed.
